addie/rietveld/braggtree.py

```diff
@@ -317,11 +345,39 @@ def do_select_gss_node(self):
         """
         # get selected nodes
         selected_nodes = self.get_selected_items()
+        selected_nodes = [str(item.text()) for item in selected_nodes]
+
+        # ZYP -> If both parent and children are selected from the tree, we want
+        # to get rid of children from the selected wks list. Meanwhile, we want
+        # to check the box corresponding to the selected children.
+        leaf_dict_temp = self._main_window.rietveld_ui.treeWidget_braggWSList._leafDict
+        for item_temp in selected_nodes:
+            if item_temp not in list(leaf_dict_temp.keys()):
+                parent_found = False
+                for key, list_temp in leaf_dict_temp.items():
+                    for item_temp_1 in list_temp:
+                        if item_temp in item_temp_1:
+                            bank_temp = list_temp.index(item_temp_1) + 1
+                            self._main_window._braggBankWidgets[bank_temp].setChecked(True)
+                            parent_found = True
+                            parent_in_tree = key
+                            break
+                    if parent_found:
+                        break
```

Comment:
Looks like this is repeated below in lines 239-255 (correct me if wrong)
See comment above at those lines for suggestion.